pkg/policy/api/rules.go

```diff
@@ -55,3 +59,20 @@ func (r Rules) String() string {
 
 	return rulesString
 }
+
+// GetSetOfLabels returns a unique set of labels defined in all rules of `rs`.
+func (rs Rules) GetSetOfLabels() []labels.LabelArray {
+	var labelSet []labels.LabelArray
+	for _, r := range rs {
+		exists := false
+		for _, set := range labelSet {
+			if r.Labels.Equals(set) {
+				exists = true
```

Comment:
Not too helpful, but a `break` here might short-circuit some of the equality checks. It makes the code more brittle, though.